pkg/asset/asset.go

```diff
@@ -118,14 +118,96 @@ type Config struct {
 	NetworkProvider        string
 	BootstrapSecretsSubdir string
 	Images                 ImageVersions
+
+	// PodCIDR describes the networking subnet to be used for inter-pod networking.
+	//
+	// Deprecated: PodCIDR exists only for compatibility with older external
+	// systems.  Please use PodCIDRs instead, which allows for dual-stack
+	// configurations.
+	PodCIDR *net.IPNet
+
+	// ServiceCIDR describes the networking subnet to be used to expose services.
+	//
+	// Deprecated: ServiceCIDR exists only for compatibility with older external
+	// systems.  Please use ServiceCIDRs instead, which allows for dual-stack
+	// configurations.  If both are specified, only ServiceCIDRs will be used.
+	ServiceCIDR *net.IPNet
+
+	// APIServiceIP describes the in-cluster IP address by which the API Servers may be reached.
+	//
+	// Deprecated: APIServiceIP exists only for compatibility with older
+	// external systems.  Please use APIServiceIPs instead, which allows for
+	// dual-stack configurations.  If both are specified, only APIServiceIPs
+	// will be used.
+	APIServiceIP net.IP
+
+	// DNSServiceIP describes the in-cluster IP address by which the cluster DNS servers may be reached.
+	//
+	// Deprecated:  DNSServiceIP exists only for compatibility with older
+	// external systems.  Please use DNSServiceIPs instead, which allows for
+	// dual-stack configurations.  If both are specified, only DNSServiceIPs
+	// will be used.
+	DNSServiceIP net.IP
+}
+
+// BindAllAddress indicates the address to use when binding all IPs.  If this
+// is an IPv6 or dual-stack system, `::` will be returned.  Otherwise
+// `0.0.0.0`.
+func (c *Config) BindAllAddress() string {
+	for _, ip := range c.APIServiceIPs {
+		if ip == nil || ip.IsLoopback() || ip.IsUnspecified() {
+			continue
+		}
+
+		if ip.To4() == nil && ip.To16() != nil {
+			return "::"
+		}
+	}
+	return "0.0.0.0"
+}
+
+// ServiceCIDRsString returns a "," concatenated string for the ServiceCIDRs
+func (c *Config) ServiceCIDRsString() string {
+	return joinStringsFromSliceOrSingle(stringerSlice(c.ServiceCIDRs), c.ServiceCIDR.String())
+}
+
+// PodCIDRsString returns a "," concatenated string for the PodCIDRs
+func (c *Config) PodCIDRsString() string {
+	return joinStringsFromSliceOrSingle(stringerSlice(c.PodCIDRs), c.PodCIDR.String())
+}
+
+// APIServiceIPsString returns a "," concatenated string for the APIServiceIPs
+func (c *Config) APIServiceIPsString() string {
+	return joinStringsFromSliceOrSingle(stringerSlice(c.APIServiceIPs), c.APIServiceIP.String())
+}
+
+// DNSServiceIPsString returns a "," concatenated string for the DNSServiceIPs
+func (c *Config) DNSServiceIPsString() string {
+	return joinStringsFromSliceOrSingle(stringerSlice(c.DNSServiceIPs), c.DNSServiceIP.String())
+}
+
+func stringerSlice(in interface{}) []string {
+	r := reflect.ValueOf(in)
+	rval := make([]string, r.Len())
+	for i := 0; i < r.Len(); i++ {
+		rval[i] = r.Index(i).Interface().(fmt.Stringer).String()
+	}
+	return rval
+}
```

Comment:
Testing this would ensure `PodCIDRsString`, `APIServiceIPsString`, and `DNSServiceIPsString` are more or less working as expected. If we could add a test for this that would be 👌 .

Reply:
Added